typescript/ws-protocol/src/FoxgloveServer.ts

```diff
@@ -145,17 +160,30 @@ export default class FoxgloveServer {
     };
 
     connection.onmessage = (event: MessageEvent<ArrayBuffer | string>) => {
-      if (typeof event.data !== "string") {
-        throw new Error(`Expected text message, got ${typeof event.data}`);
-      }
-      const message = JSON.parse(event.data) as unknown;
-      if (typeof message !== "object" || message == undefined) {
-        throw new Error(`Expected JSON object, got ${typeof message}`);
-      }
-      try {
-        this.handleClientMessage(client, message as ClientMessage);
-      } catch (error) {
-        this.emitter.emit("error", error as Error);
+      if (typeof event.data === "string") {
+        // TEXT (JSON) message handling
+        const message = JSON.parse(event.data) as unknown;
```

Comment:
Not new in your PR, but it occurs to me that maybe the `JSON.parse` should be inside a try/catch as well. We might end up with an uncaught exception if the client sends invalid json.